lib/AST/Type.cpp

```diff
@@ -75,7 +75,7 @@ Type QuerySubstitutionMap::operator()(SubstitutableType *type) const {
 }
 
 void TypeLoc::setType(Type Ty) {
-  assert(!Ty || !Ty->hasTypeVariable());
+//  assert(!Ty || !Ty->hasTypeVariable());
```

Comment:
I'm not happy about this assertion being removed. If it means we have to scrub TypeLoc in order to merge this, I think we're down to enough places that we should do that first.

Reply:
This assertion caught me by surprise. Is there a general idea behind thinning out TypeLoc? That is, do we have a replacement?

Reply:
Never mind actually, I will be fine by just looking at your April TypeLoc cleanups.